src/components/alerts/WalletAlert.tsx

```diff
@@ -60,28 +106,122 @@ const WalletAlert = ({ isOpen, onCancel, refetch }: Props) => {
           ) {
             toast({
               title: "Error",
-              description: response.response.data.error,
+              description:
+                response.response?.data?.message || "Something went wrong!",
               status: "error",
               duration: 9000,
               isClosable: true,
             });
             return;
           }
-
-          handleClose();
-          refetch();
-          toast({
-            title: "Paid",
-            description: "We've paid your invoice. Keep reviewing!",
-            status: "success",
-            duration: 9000,
-            isClosable: true,
-          });
+          if (response.metadata) {
+            const metadataArray = JSON.parse(response.metadata);
+            const addressPosition =
+              metadataArray
+                .flat()
+                .findIndex((elm: string) => elm.includes("text/identifier")) +
+              1;
+            if (metadataArray.flat()[addressPosition] === invoiceInput) {
+              setLightningData(response as LightningResponse);
+              setStep(2);
+            }
+            return;
+          }
         },
       }
     );
   };
 
+  const handleWithdrawAmount = (e: ChangeEvent<HTMLInputElement>) => {
+    let sendingAmount = e.target.value;
+
+    if (sendingAmount.match(/^[0-9]+$/)) {
+      // turn into a number type
+      setAmountToSend(sendingAmount);
+    } else {
+      if (sendingAmount.length === 0) {
+        setAmountToSend("");
+      }
+    }
+  };
+
+  const handleWithdrawalValidation = () => {
+    const amountToSendParsed = +(amountToSend || 0);
+    const isWithinBalance = +amountToSendParsed <= +balance;
+    const minimumSendable = (lightningData?.minSendable || 1000) / 1000;
+    const maximumSendable = (lightningData?.maxSendable || 1000) / 1000;
+    const isWithinSendable =
+      +amountToSendParsed >= minimumSendable &&
+      amountToSendParsed <= maximumSendable;
+    setError("");
+    if (!isWithinBalance) {
+      setError("Error: send sats lower than balance");
+      return;
+    }
+
+    if (!isWithinSendable) {
+      setError("Error: send sats within range");
+      return;
+    }
+    payInvoice.mutate(
+      {
+        amount: amountToSend as string,
+        callbackUrl: lightningData?.callback as string,
+      },
+      {
+        onSuccess: (response) => {
+          if (
+            response?.response?.status >= 400 ||
+            response?.response?.status <= 500
+          ) {
+            toast({
+              title: "Error",
+              description:
+                response.response?.data?.message || "Something went wrong!",
+              status: "error",
+              duration: 9000,
+              isClosable: true,
+            });
+            return;
+          }
+          withdrawSats.mutate(
+            {
+              invoice: response?.pr || "",
```

Comment:
You can add a check for `payment request` before getting here so you don't have to use an `|| ""`.

There's a check above, you can include it there

Reply:
so an if statement for if(response.pr)

Reply:
there's already a check above. you can include it there.

So at the point you get here, typescript knows it is a defined value

Reply:
alright 👍